pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -163,6 +165,17 @@ private String executeSqlQuery(@Context HttpHeaders httpHeaders, String sqlQuery
     } catch (SqlCompilationException ex) {
       throw QueryException.getException(QueryException.SQL_PARSING_ERROR, ex);
     }
+
+    String tableName = CalciteSqlParser.compileToPinotQuery(sqlQuery).getDataSource().getTableName();
```

Comment:
This will only work if the user issues a query via the controller query API. The recommended way of querying Pinot is via the broker query API (https://docs.pinot.apache.org/users/api/querying-pinot-using-standard-sql) which is also what the controller API delegates to - https://github.com/apache/pinot/blob/c6c0fa243b7be407c37749e45512f156297e271a/pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java#L355

Ideally, we'd want this check in the broker request handler, since all external REST API based queries will always go through there.

Reply:
Gotcha, I've moved it to the broker request handler right after we check if the table exists.

Reply:
@yashmayya Gotcha. I'm looking at the `isTableEnabled` function in the `PinotHelixResourceManager`, but can't quite figure out how to use that in the broker request handler. Is there something else I should use instead?

Reply:
The `BaseBrokerStarter` which instantiates the single-stage broker request handler and the multi-stage broker request handler also has a `HelixAdmin` that can be used to check the table's ideal state in ZK (which has the enabled / disabled information). So you could inject this admin (and the cluster name) as a dependency into the broker request handlers from the `BaseBrokerStarter` and use it to check whether the tables in a request are disabled.